mne/utils/config.py

```diff
@@ -358,9 +358,12 @@ def get_subjects_dir(subjects_dir=None, raise_error=False):
     value : str | None
         The SUBJECTS_DIR value.
     """
+    _validate_type(item=subjects_dir, types=('path-like', None),
+                   item_name='subjects_dir', type_name='str or path-like')
+
     if subjects_dir is None:
         subjects_dir = get_config('SUBJECTS_DIR', raise_error=raise_error)
-    return subjects_dir
+    return str(subjects_dir)
```

Comment:
This can end up being `'None'` whereas before it could be `None`:
```
$ _MNE_FAKE_HOME_DIR=$PWD python -c "import mne; print(mne.utils.get_subjects_dir(None, raise_error=False))"
None
```
I think you need to protect the cast with a `if subjects_dir is not None`
```suggestion
    if subjects_dir is not None:
        subjects_dir = str(subjects_dir)
    return subjects_dir
```

Reply:
(please also add a regression test for this if `pytest mne/utils` passes on your system without this change)

Reply:
But we already check for `None` in line 364, am I missing something?

Reply:
Hm okay seems one test is failing, never mind. I will look into this.

Reply:
`get_config` can return None (see my example above where it does this)

Reply:
Got it, thanks